infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -110,6 +110,43 @@ public LogReplicationMetadataManager(CorfuRuntime rt, long topologyConfigId, Str
         setupTopologyConfigId(topologyConfigId);
     }
 
+    public void initializeReplicationStatusTable(String remoteClusterId) {
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
+                    ReplicationStatusKey replicationStatusKey = ReplicationStatusKey.newBuilder()
+                            .setClusterId(remoteClusterId)
+                            .build();
+
+                    // Only set the default value if the key is not present
+                    if (!txn.isExists(replicationStatusTable, replicationStatusKey)) {
+                        ReplicationStatusVal defaultSourceStatus = ReplicationStatusVal.newBuilder()
+                                .setStatus(SyncStatus.NOT_STARTED)
+                                .setRemainingEntriesToSend(-1L)
```

Comment:
A couple of questions here:
* I remember the purpose of this `isExists` check is because all the nodes will try to initialize status table (not only leader node). But in real deployment will it happen that there exists a stale entry in persisted status table? (like from LR happened long time ago between these 2 clusters) - in this case the semantics is different right? key exists because other node already initialized status table vs key exists because stale data is read
* When initialize status table here we implicitly set `SyncType` to be `SNAPSHOT`, should we initialize the in-memory `lastSyncType` here as well?
* Further, similar to another comment in `InitializedState.java`, I found maintaining the in-memory `lastSyncType` introduces additional confusion and complexity. Ideally should we just read the status table instead?

Reply:
Hi @xcchang 

- The stale entry case is very rare because the status table is cleared upon a role change and whenever the standby clusters change, the old ones are removed.  If someone shut down a cluster abruptly and repurposed it without cleaning the stale data, it may happen.  But such a workflow is not likely IMO.
- Any benefit of initializing lastSyncType here?
- Just reading the status table in all cases is not possible because the cluster may have been shut down for a long time, the table can reflect a log entry sync but after a restart, a snapshot sync is required.  The FSM is required to take this decision.

Reply:
for 3), do you mean that the fsm state transitions should write the sync type and the AckReader just updates the status?

Reply:
Hi Pankti, thanks! I see the point for (1). For (2) and (3) actually my confusion came from that in-memory `lastSyncType` field. I replied to your another comment: https://github.com/CorfuDB/CorfuDB/pull/3570#discussion_r1154833946